test/sqllogictest/temporal.slt

```diff
@@ -0,0 +1,45 @@
+# Copyright Materialize, Inc. All rights reserved.
```

Comment:
Due to the fact that GitHub only allows resolvable comments in files and not in the general comment section, I will add my comments to the test file instead.

Item 1. The restriction against not being able to use NotEq is not enforced. This query is accepted:
```
select * from intervals where mz_logical_timestamp() <> 100;
 a | b  
---+----
 1 |  2
 1 | 10
 2 | 13
 3 |  1
(4 rows)
```

Reply:
Same for the restriction that ```mz_logical_timestamp``` can not appear on both sides of an expression.

Reply:
Interesting. I've certainly *failed* that test in other code. I must have broken it since! I'll check both of these out (there is a method `extract_temporal` that is meant to error on such; I'll leave some tests closer to it as well).

Reply:
Ah, I see what is going on here. This is a docs issue I think.

When you directly invoke `select` you perform a "one-shot" query, and we can accept all uses of `mz_logical_timestamp()`. We just replace it with the actual logical timestamp and treat it as a constant. If you try to create or materialize a view that uses this fragment it should error:
```
materialize=> CREATE VIEW test AS select * from intervals where mz_logical_timestamp() <> 100;ERROR:  "Unsupported binary temporal operation"
materialize=> 
```

Reply:
Thank you, understood.